cabal-install/src/Distribution/Client/CmdOutdated.hs

```diff
@@ -1,510 +1,199 @@
+    when (isJust mprojectDir || isJust mprojectFile) $
+      dieWithException verbosity OutdatedAction
 
-defaultOutdatedFlags :: OutdatedFlags
-defaultOutdatedFlags =
-  OutdatedFlags
-    { outdatedVerbosity = toFlag normal
-    , outdatedFreezeFile = mempty
-    , outdatedNewFreezeFile = mempty
-    , outdatedSimpleOutput = mempty
-    , outdatedExitCode = mempty
-    , outdatedQuiet = mempty
-    , outdatedIgnore = mempty
-    , outdatedMinor = mempty
-    }
+    sourcePkgDb <- IndexUtils.getSourcePackages verbosity repoContext
 
-outdatedOptions :: ShowOrParseArgs -> [OptionField OutdatedFlags]
-outdatedOptions _showOrParseArgs =
-  [ optionVerbosity
-      outdatedVerbosity
-      (\v flags -> flags{outdatedVerbosity = v})
-  , option
-      []
-      ["freeze-file", "v1-freeze-file"]
-      "Act on the freeze file"
-      outdatedFreezeFile
-      (\v flags -> flags{outdatedFreezeFile = v})
-      trueArg
-  , option
-      []
-      ["v2-freeze-file", "new-freeze-file"]
-      "Act on the new-style freeze file (default: cabal.project.freeze)"
-      outdatedNewFreezeFile
-      (\v flags -> flags{outdatedNewFreezeFile = v})
-      trueArg
-  , option
-      []
-      ["simple-output"]
-      "Only print names of outdated dependencies, one per line"
-      outdatedSimpleOutput
-      (\v flags -> flags{outdatedSimpleOutput = v})
-      trueArg
-  , option
-      []
-      ["exit-code"]
-      "Exit with non-zero when there are outdated dependencies"
-      outdatedExitCode
-      (\v flags -> flags{outdatedExitCode = v})
-      trueArg
-  , option
-      ['q']
-      ["quiet"]
-      "Don't print any output. Implies '--exit-code' and '-v0'"
-      outdatedQuiet
-      (\v flags -> flags{outdatedQuiet = v})
-      trueArg
-  , option
-      []
-      ["ignore"]
-      "Packages to ignore"
-      outdatedIgnore
-      (\v flags -> flags{outdatedIgnore = v})
-      (reqArg "PKGS" pkgNameListParser (map prettyShow))
-  , option
-      []
-      ["minor"]
-      "Ignore major version bumps for these packages"
-      outdatedMinor
-      (\v flags -> flags{outdatedMinor = v})
-      ( optArg
-          "PKGS"
-          ignoreMajorVersionBumpsParser
-          ("", Just IgnoreMajorVersionBumpsAll)
-          ignoreMajorVersionBumpsPrinter
-      )
-  ]
-  where
-    ignoreMajorVersionBumpsPrinter
-      :: Maybe IgnoreMajorVersionBumps
-      -> [Maybe String]
-    ignoreMajorVersionBumpsPrinter Nothing = []
-    ignoreMajorVersionBumpsPrinter (Just IgnoreMajorVersionBumpsNone) = []
-    ignoreMajorVersionBumpsPrinter (Just IgnoreMajorVersionBumpsAll) = [Nothing]
-    ignoreMajorVersionBumpsPrinter (Just (IgnoreMajorVersionBumpsSome pkgs)) =
-      map (Just . prettyShow) pkgs
+    prjBaseCtxt <- establishProjectBaseContext verbosity cliConfig OtherCommand
 
-    ignoreMajorVersionBumpsParser =
-      (Just . IgnoreMajorVersionBumpsSome) `fmap` pkgNameListParser
+    pkgSpecifiers <-
```

Comment:
isn't this the same as `localPackages` in `prjBaseCtxt`?

Reply:
You are correct. Simplified it by removing the call to `fetchAndReadSourcePackages`.